hollow/src/main/java/com/netflix/hollow/api/codegen/api/HollowDataAccessorGenerator.java

```diff
@@ -0,0 +1,114 @@
+
+import com.netflix.hollow.api.codegen.HollowAPIGenerator;
+import com.netflix.hollow.api.codegen.HollowJavaFileGenerator;
+import com.netflix.hollow.api.consumer.HollowConsumer;
+import com.netflix.hollow.api.consumer.data.AbstractHollowDataAccessor;
+import com.netflix.hollow.api.custom.HollowAPI;
+import com.netflix.hollow.core.index.HollowPrimaryKeyIndex;
+import com.netflix.hollow.core.index.key.PrimaryKey;
+import com.netflix.hollow.core.read.engine.HollowReadStateEngine;
+import com.netflix.hollow.core.schema.HollowObjectSchema;
+import com.netflix.hollow.core.write.objectmapper.HollowPrimaryKey;
+
+/**
+ * This class contains template logic for generating a {@link HollowAPI} implementation. Not intended for external consumption.
+ *
+ * @see HollowAPIGenerator
+ */
+public class HollowDataAccessorGenerator implements HollowJavaFileGenerator {
+
+    protected final String packageName;
+    protected final String classname;
+    protected final String apiClassname;
+    protected final String classPostfix;
+    protected final String type;
+    protected final boolean useAggressiveSubstitutions;
+    protected final HollowObjectSchema schema;
+
+    public HollowDataAccessorGenerator(String packageName, String apiClassname, String classPostfix, boolean useAggressiveSubstitutions, HollowObjectSchema schema) {
+        this.classname = getClassName(schema);
+        this.apiClassname = apiClassname;
+        this.classPostfix = classPostfix;
+        this.packageName = packageName;
+        this.type =  hollowImplClassname(schema.getName(), classPostfix, useAggressiveSubstitutions);
+        this.useAggressiveSubstitutions = useAggressiveSubstitutions;
+        this.schema = schema;
+    }
+
+    protected String getClassName(HollowObjectSchema schema) {
+        return schema.getName() + "DataAccessor";
+    }
+
+    @Override
+    public String getClassName() {
+        return classname;
+    }
+
+    @Override
+    public String generate() {
+        StringBuilder builder = new StringBuilder();
+
+        builder.append("package " + packageName + ";\n\n");
+
+        builder.append("import " + HollowConsumer.class.getName() + ";\n");
+        builder.append("import " + AbstractHollowDataAccessor.class.getName() + ";\n");
+        builder.append("import " + PrimaryKey.class.getName() + ";\n");
+        builder.append("import " + HollowReadStateEngine.class.getName() + ";\n");
+
+        builder.append("\n");
+        builder.append("public class " + classname + " extends " + AbstractHollowDataAccessor.class.getSimpleName() + "<" + type  +"> {\n\n");
+
+        builder.append("    private " + apiClassname + " api;\n\n");
+
+        genConstructors(builder);
+        genPublicAPIs(builder);
+
+        builder.append("}");
+
+        return builder.toString();
+    }
+
+    protected void genConstructors(StringBuilder builder) {
+        builder.append("    public " + classname + "(HollowConsumer consumer, String type) {\n");
```

Comment:
Why do we need to specify a type here -- isn't the type implied based on the schema from which this is generated?

Reply:
You are right, I don't need that in generated code, I was just blindly overriding base class ;)

Good catch.